src/transformers/models/llama/modeling_llama.py

```diff
@@ -974,13 +974,16 @@ def forward(
             inputs_embeds = self.embed_tokens(input_ids)
 
         past_seen_tokens = 0
+        is_static_cache = isinstance(past_key_values, StaticCache) or isinstance(
+            getattr(self.layers[0].self_attn, "past_key_value", None), StaticCache
+        )
```

Comment:
Cc @gante in this release I think we can already not use any of the logic that does rely on the `DynamicCache` in favor of relying on the cache positions WDYT?

Reply:
@ArthurZucker what do you mean by "logic that does rely on the DynamicCache"? The `past_seen_tokens = past_key_values.get_seq_length()` line?

Reply:
Yes, and the max_length as well! I just don't want to add a new variable that is not general an is only used in _update_causal_mask.